### PR TITLE
Peer monitor for geth

### PR DESCRIPTION
**Pull request by tailingchen**
This PR implements a peer monitor to check eth node's peer set and add remote nodes as static node if peer count is not enough.

### CODE REVIEW

File: cmd/peer-monitor/ethclient.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// NewClient creates a client that uses the given RPC client.
+func NewClient(c *rpc.Client) *Client {
+	return &Client{c}
+}
+
+func (ec *Client) Close() {
+	ec.c.Close()
+}
+
+func (ec *Client) Peers() ([]*p2p.PeerInfo, error) {
```
Comment (markya0616) [inline cmd/peer-monitor/ethclient.go]: How about pass ctx in and set timeout in outside?

File: cmd/peer-monitor/ethclient.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+	var result []*p2p.PeerInfo
+	err := ec.c.CallContext(ctx, &result, "admin_peers")
+	return result, err
+}
+
+func (ec *Client) AddPeer(url string) error {
```
Comment (markya0616) [inline cmd/peer-monitor/ethclient.go]: ditto

File: cmd/peer-monitor/root.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initViper)
+
+	ServerCmd.AddCommand(onceCmd)
+
+	// eth-client flags
+	ServerCmd.PersistentFlags().String(ethURLFlag, "ipc://data/geth.ipc", "The url of eth server")
```
Comment (markya0616) [inline cmd/peer-monitor/root.go]: -> `/data/geth.ipc`

File: cmd/peer-monitor/ethclient.go (hunk 1 above)
Comment (tailingchen) [reply cmd/peer-monitor/ethclient.go]: Updated in `acdb03c`.

File: cmd/peer-monitor/ethclient.go (hunk 2 above)
Comment (tailingchen) [reply cmd/peer-monitor/ethclient.go]: Updated in `acdb03c`.

File: cmd/peer-monitor/monitor.go [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+		return nil
+	}
+
+	nodes := fetchNodes(peers, m.maxPeerCount)
+	if len(nodes) == 0 {
+		return errors.New("empty node list")
+	}
+	for _, n := range nodes {
+		err := ethClient.AddPeer(n)
+		log.Trace("Add peer", "node", n, "err", err)
```
[the remark this answers is not in the document]
Comment (tailingchen) [reply cmd/peer-monitor/monitor.go]: True, should use batch request here.
Updated in `acdb03c`.

File: cmd/peer-monitor/root.go (hunk 3 above)
Comment (tailingchen) [reply cmd/peer-monitor/root.go]: Thanks!
Updated in `acdb03c`.

**Pull request merged**